### PR TITLE
Stable rust

### PR DESCRIPTION
**Pull request by mattciferri**
### Context

Hecate currently runs on a specific version of the nightly build of Rust, largely due to the web framework we ran on, Rocket. Now that we've [cut over to Actix](https://github.com/mapbox/Hecate/pull/180), we can build off of Rust stable. This brings a more consistent experience for those working on Hecate, and allows us to leverage the tried and true features of rust without worrying about the instability of nightly.

## Summary of changes

- [x] Removes the nightly toolchain flag from the install script.
- [x] Removes json-patch.
- [x] Removes unnecessary deps from the build steps.
- [x] Moves the ci subproject from npm to yarn.

## Next steps

- [x] Test that CI can build dev and release versions, and passes tests.
- [x] Test on a staging stack. 

cc @ingalls @lizziegooding @miccolis @samely

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -47,7 +44,7 @@ jobs:
                 name: "Install rust"
                 command: |
                   curl https://sh.rustup.rs -sSf > /tmp/rustup.sh \
-                  && sh /tmp/rustup.sh -y --default-toolchain nightly-2019-06-01 \
+                  && sh /tmp/rustup.sh -y \
```
[the remark this answers is not in the document]
Comment (ingalls) [reply .circleci/config.yml]: Edit: per below let's lock it to https://github.com/mapbox/Hecate/pull/194/files#diff-04c6e90faac2675aa89e2176d2eec7d8R76

File: README.md [hunk 2]
```diff
@@ -86,13 +86,6 @@ source ~/.bashrc        # Most Linux Distros, some OSX
 source ~/.bash_profile  # Most OSX, some Linux Distros
 ```
 
-- Install the `nightly-2019-06-01` build of rust, `Rocket`, the web-framework relies on some advanced compiler options not yet included in the default build.
```
Comment (ingalls) [inline README.md:89]: Can we not delete this and instead have install instructions for 1.38.0, saying something like "hecate should work on stable versions of rust but is tested with x"

Comment (mattciferri) [reply README.md:89]: Did you see the change above this? I added that note to the initial rust installation instructions.

Comment (ingalls) [reply README.md:89]: I did, but not everyone is going to know how to actually install it.

Comment (ingalls) [reply README.md:89]: We could include this in another line with those instructions?

**Review by ingalls — APPROVED** [review]
:taco:

**Pull request merged**